crates/api_common/src/utils.rs

```diff
@@ -668,10 +669,14 @@ pub async fn purge_image_posts_for_community(
 }
 
 pub async fn remove_user_data(
```

Comment:
Is there any difference between this function and `restore_user_data` besides the boolean value that's assigned to `removed`? If not, you could make a function that's local to this module that handles the cases and accepts the `removed` bool as a parameter. e.g. a function with a signature like:

```rust
async fn remove_or_restore_user_data(
    mod_person_id: PersonId,
    banned_person_id: PersonId,
    reason: Option<&str>,
    context: &LemmyContext,
    removed: bool
) -> LemmyResult<()> {
   // Put logic here
}
```

And then remove/restore data can just be this:

```rust
pub async fn remove_user_data(
    mod_person_id: PersonId,
    banned_person_id: PersonId,
    reason: Option<&str>,
    context: &LemmyContext
) -> LemmyResult<()> {
   remove_or_restore_user_data(mod_person_id, banned_person_id, reason, context, true).await
}
```

You could also use a const generic if you prefer:

```rust
async fn remove_or_restore_user_data<const REMOVED: bool>(
    mod_person_id: PersonId,
    banned_person_id: PersonId,
    reason: Option<&str>,
    context: &LemmyContext
) -> LemmyResult<()> {
   // Put logic here
}
```

And the public function would be:
```rust
pub async fn remove_user_data(
    mod_person_id: PersonId,
    banned_person_id: PersonId,
    reason: Option<&str>,
    context: &LemmyContext
) -> LemmyResult<()> {
   remove_or_restore_user_data::<true>(mod_person_id, banned_person_id, reason, context).await
}
```

Reply:
If you read through those two functions, they're different enough that it makes sense to keep them separate. The remove user data does a few extra things that the restore can't, like:

- purging their images
- clearing their profile images
- removing communities where they're the top mod

Whereas the restore can really only restore their posts and comments.

Reply:
Should be easy to handle those inside an `if is_remove` block.

Reply:
Alright I'll give that a try.